### PR TITLE
ARQ-2039 don't load excluded classes.

### PR DESCRIPTION
**Pull request by tremes**
This change is [Reviewable](https://reviewable.io/reviews/arquillian/arquillian-container-weld/29)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,8 +32,8 @@
         <!-- Versioning -->
         <version.arquillian_core>1.1.11.Final</version.arquillian_core>
         
-        <weld.api.version>2.3.SP2</weld.api.version>
-        <weld.core.version>2.3.5.Final</weld.core.version>
+        <weld.api.version>2.4.CR1</weld.api.version>
+        <weld.core.version>2.4.0.CR1</weld.core.version>
```
Comment (johnament) [inline pom.xml]: Why upgrade to a non-final release?

Comment (tremes) [inline pom.xml]: I am using `org.jboss.weld.metadata.FilterPredicate` which was changed in this latest release. Besides that I still plan to release next Beta or CR.

Comment (johnament) [inline pom.xml]: So then no one who uses an in-production release of weld (e.g. ships with wildfly) can leverage this extension?

Comment (tremes) [inline pom.xml]: Sorry I don't follow. You can either use current `Beta2` version (with 2.3.5.Final) or wait for next release with newer Weld version (I think I can wait for 2.4.0.Final).

Comment (tremes) [inline pom.xml]: Not updating to latest Weld version would mean that you can't use the upcoming version with the Weld latest version. So I think it's better approach to stick to the latest version.

Comment (bartoszmajsak) [inline pom.xml]: I'm not sure I get the concern to its full extent, but if I understand @johnament it means that he, while being on Weld 2.3.x.Final, wouldn;t be able to use Weld Container adapter 2.0.x as it's build against newer version of Weld. Maybe we should make versioning more aligned with Weld versions then? Would that solve anything?

Comment (tremes) [inline pom.xml]: Basically yes. As I said I can release next version of `arquillian-container-weld` once there is available next stable (Final) version of Weld but currently I don't consider this as an obstacle in merging this PR.

Comment (mkouba) [inline pom.xml]: Just a note - I think `arquillian-container-weld` 2.x should still work with Weld 2.3.x unless the functionality added in this PR (processing exclude filters from `beans.xml`) is used.

Comment (johnament) [inline pom.xml]: Not sure what you mean, `weld:scan` has been around since 1.1.

Comment (mkouba) [inline pom.xml]: Yep, but it was never implemented in this adapter (@tremes correct me if I'm wrong). I'm talking about standardized version of `weld:scan` - 12.4.2. Exclude filters (CDI 1.1+).

Comment (tremes) [inline pom.xml]: You are correct. It wasn't implemented. Are there any objections to merge this PR in its current shape?

Comment (bartoszmajsak) [inline pom.xml]: @johnament are you fine with it?

**Comment by tremes** [comment]
Updated. John is right that it's better to update Weld version with https://github.com/arquillian/arquillian-container-weld/pull/30

**Comment by tremes** [comment]
Merged.

**Pull request closed without merging**